packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -915,10 +919,42 @@ actual object RealmInterop {
                 }
             },
             StableRef.create(callback).asCPointer(),
-            staticCFunction { userdata -> disposeUserData<Callback>(userdata) }
+            staticCFunction { userdata -> disposeUserData<AppCallback<NativePointer>>(userdata) }
         )
     }
 
+    actual fun realm_app_log_out(app: NativePointer, user: NativePointer, callback: AppCallback<Unit>) {
+        realm_wrapper.realm_app_log_out(
+            app.cptr(),
+            user.cptr(),
+            staticCFunction { userdata, error: CPointer<realm_app_error_t>? ->
+                val userDataCallback = safeUserData<AppCallback<Unit>>(userdata)
+                if (error == null) {
+                    userDataCallback.onSuccess(Unit)
+                } else {
+                    val message = with(error.pointed) {
+                        "${message?.toKString()} [error_category=${error_category.value}, error_code=$error_code, link_to_server_logs=$link_to_server_logs]"
+                    }
+                    userDataCallback.onError(AppException(message))
```

Comment:
This was added to `cinterop` in the initial implementation, which was technically sufficient before we had docs as `cinterop` is an `api`-dependency. We don't generate dokka for this, so it will not be part of the documentation, and agree that we should get to a point where there is no public API in `cinterop`. 

Putting `AppException` in `library-sync` will require us to have a separate cinterop-exception, that is caught and rethrown in `library-sync`. This specifically ties into the issue of proper abstraction for all our error codes of #522, so an initial layering cinterop-exception -> AppException is probably just one step towards making an appropriate mapping in `library-sync`. 

I would argue that this rework should go into another PR as it also relates to `login`.